SonarQube.TeamBuild.Integration/CoverageReportProcessorBase.cs

```diff
@@ -29,36 +34,47 @@ protected CoverageReportProcessorBase(ICoverageReportConverter converter)
 
         #region ICoverageReportProcessor interface
 
-        public bool ProcessCoverageReports(AnalysisConfig context, TeamBuildSettings settings, ILogger logger)
+
+        public bool Initialise(AnalysisConfig config, TeamBuildSettings settings, ILogger logger)
         {
-            if (context == null)
+            if (config == null)
+            {
+                throw new ArgumentNullException("config");
+            }
+            if (settings == null)
             {
-                throw new ArgumentNullException("context");
+                throw new ArgumentNullException("settings");
             }
             if (logger == null)
             {
                 throw new ArgumentNullException("logger");
             }
 
-            if (!this.converter.Initialize(logger))
-            {
-                // If we can't initialize the converter (e.g. we can't find the exe required to do the
-                // conversion) there in there isn't any point in downloading the binary reports
-                return false;
-            }
+            this.config = config;
+            this.settings = settings;
+            this.logger = logger;
+
+            this.succesfullyInitialised =  this.converter.Initialize(logger);
+            return succesfullyInitialised;
+        }
+
+        public bool ProcessCoverageReports()
+        {
+            Debug.Assert(this.config != null, "Call initialise first");
+            Debug.Assert(this.succesfullyInitialised, "Initialisation failed, cannot process coverage reports");
```

Comment:
Throw an InvalidOperationException if this.successfullyInitialised is false, as this is checking that external use of this class at runtime is correct.

Asserting that config is not null is ok, but the assertion message isn't appropriate. It's checking an internal assumption about the implementation of this class i.e. that the Initialise method has checked the config is not null and stored it.
The assertion should be after the exception.